app/controllers/posts_controller.rb

```diff
@@ -1,36 +1,34 @@
 class PostsController < ApplicationController
   def index
-    @user = User.find(params[:user_id])
-    @posts = @user.posts
-  end
-
-  def show
-    @post = Post.find(params[:id])
+    @posts = Post.all
+    @user = User.find(params[:user_id].to_i)
```

Comment:
Kindly make sure that the app solves the N+1 problem when fetching all posts and their comments for a user as per project requirements.

For example in the `posts#index` I see that your previous code was
```
  def index
    @user = User.find(params[:user_id])
    @posts = @user.posts
  end
```

So what you could have done was use `@posts = @user.posts.includes(:comments)` instead of `@posts = @user.posts` to solve the N+1 problem .